public/src/components/teacher/GroupResults.tsx

```diff
@@ -0,0 +1,119 @@
+interface IResultsProp {
+    course: ICourse;
+    groups: IGroupCourseWithGroup[];
+    labs: IAssignment[];
+    onApproveClick: (submissionID:number) => void;
+}
+interface IResultsState {
+    assignment?: IStudentSubmission;
+    groups: IGroupCourseWithGroup[];
+}
+class GroupResults extends React.Component<IResultsProp, IResultsState> {
+    constructor(props: IResultsProp) {
+        super(props);
+
+        
+        if (this.props.groups[0] && this.props.groups[0].course.assignments[0]) {
+            this.state = {
+                assignment: this.props.groups[0].course.assignments[0],
+                groups: this.props.groups,
+                
+            };
+        } else {
+            this.state = {
+                assignment: undefined,
+                groups: this.props.groups,
+            };
+        }
+    }
+
+    public render() {
+        let groupLab: JSX.Element | null = null;
+        let tmp: JSX.Element | null = null;
+        
+        if (this.props.groups 
+            && this.props.groups.length > 0 
+            && this.state.assignment
+            && this.state.assignment.assignment.isgrouplab) 
+            {
+                groupLab = <StudentLab
+                course={this.props.course}
+                assignment={this.state.assignment}
+                showApprove={true}
+                onRebuildClick={() => {}}
+                onApproveClick={() => {
+                    if (this.state.assignment && this.state.assignment.latest) {
+                        this.props.onApproveClick(this.state.assignment.latest.id)
+                    }
+                }}
+            />;
+
+        }
+
+        return (
+            <div>
+                <h1>Result: {this.props.course.name}</h1>
+                <Row>
+                    <div className="col-lg6 col-md-6 col-sm-12">
+                        <Search className="input-group"
+                            placeholder="Search for groups"
+                            onChange={(query) => this.handleOnchange(query)}
+                        />
+                        <DynamicTable header={this.getResultHeader()}
+                            data={this.state.groups}
+                            selector={(item: IGroupCourseWithGroup) => this.getGroupResultSelector(item)} 
+                        />
+                        {tmp}
+                    </div>
+                    <div className="col-lg-6 col-md-6 col-sm-12">
+                        {groupLab}
+                    </div>
+                </Row>
+            </div>
+        );
+    }
+
+    private getResultHeader(): string[] {
+        let headers: string[] = ["Name", "Slipdays"];
+        headers = headers.concat(this.props.labs.filter((e) => e.isgrouplab).map((e) => e.name));
+        return headers;
+    }
+
+    private getGroupResultSelector(group: IGroupCourseWithGroup): Array<string | JSX.Element> {
+        let selector: Array<string | JSX.Element> = [group.group.name, "5"];
```

Comment:
What does `5` mean?

Reply:
This is how slip days are posted to the user, right now they are hardcoded, both for Results and GroupResults, so need to implement some slip day logic for front-end and backend.

Reply:
Ah, ok. Suggest to make it into a variable (or temporary constant) so that it is easier to recall, when we get time to implement the necessary logic.